src/gui/widgets/options_button.cpp

```diff
@@ -202,27 +254,63 @@ void options_button::set_values(const boost::container::stable_vector<menu_item>
 		}
 	}
 
-	for (auto& item: values) {
-		values_.emplace_back(item.get_config());
+    toggle_states_.resize(values.size(), false);
+    toggle_states_.reset();
+
+	for (std::size_t i = 0; i < values.size(); i++) {
+		values_.emplace_back(values[i].get_config());
+		if((values[i].checkbox) && (values[i].checkbox.value())) {
+			toggle_states_.set(i);
+		}
 	}
 
 	selected_ = selected;
 
 	if(persistent_) {
 		set_label(values_[selected_].label);
 	}
+
 }
 
-menu_item& options_button::add_row(const config& row, const int index)
+menu_item& options_button::add_row(config row, const int index)
 {
+	bool appended = false;
+
 	if(index < -1) {
 		assert(index >= -1); // this looks odd, but sends right message to user
-	} else if((index == -1) || (static_cast<size_t>(index) == values_.size())) {
-		values_.emplace_back(row);
+	}
+
+	if((get_control_type() == "multimenu_button") &&
```

Comment:
This code belongs in the multimenu_button. Make `add_row` virtual if it's not already, then add a function n multimenu_button that first does this and then does `return menu_item::add_row(row, index);`

Reply:
Yeah.  There was some other stuff before that at one point and I wasn't sure how to handle that case.